uninstall.sh

```diff
@@ -0,0 +1,350 @@
+  for s in "$@"; do
+    s=${s## }
+    echo "${s%% }"
+  done
+}
+
+Dir_children() {
+  local p
+  for p in "$@"; do
+    [[ -d $p ]] || continue
+    find "$p" -maxdepth 1 | grep -v "^${p}$"
+  done
+}
+
+###
+
+# Default options
+opt_force=""
+opt_quiet=""
+opt_dry_run=""
+opt_skip_cache_and_logs=""
+
+# global status to indicate whether there is anything wrong.
+failed=false
+
+un=$(uname)
+case "$un" in
+  Linux)
+    ostype=linux
+    homebrew_prefix_default=/home/linuxbrew/.linuxbrew
+  ;;
+  Darwin)
+    ostype=macos
+    homebrew_prefix_default=/usr/local
+    # Stock macOS doesn't have a realpath, so we make do with builtin Python
+    realpath() {
+      /usr/bin/python -c 'import os,sys;print(os.path.realpath(sys.argv[1]))' "$1"
+    }
+  ;;
+  *)
+    Kernel_abort "Unsupported system type '$un'"
+  ;;
+esac
+
+# string formatters
+if [[ -t 1 ]]; then
+  Tty_escape() { printf "\033[%sm" "$1"; }
+else
+  Tty_escape() { :; }
+fi
+Tty_mkbold() { Tty_escape "1;${1:-39}"; }
+Tty_blue=$(Tty_mkbold 34)
+Tty_red=$(Tty_mkbold 31)
+Tty_bold=$(Tty_mkbold 39)
+Tty_reset=$(Tty_escape 0)
+
+Array_shell_s() {
+  local result=("$1"); shift
+  local s
+  for s in "$@"; do
+    result+=("${s// /\\ }")
+  done
+  echo -n "${result[*]}"
+}
+
+Pathname_resolved_path() { realpath "$1"; }
+
+Pathname_pretty_print() {
+  local p
+  for p in "$@"; do
+    if [[ -h $p ]]; then
+      printf '%s -> %s\n' "$p" "$(Pathname_resolved_path "$p")"
+    elif [[ -d $p ]]; then
+      echo "$p/"
+    else
+      echo "$p"
+    fi
+  done
+}
+
+ohai() { echo "${Tty_blue}==>${Tty_bold} $(Array_shell_s "$@")${Tty_reset}"; }
+
+warn() { echo "${Tty_red}Warning${Tty_reset}: ${1##$'\n'}"; }
+
+system() {
+  if ! "$@"; then
+    warn "Failed during: $(Array_shell_s "$@")"
+    failed=true
+  fi
+}
+
+####################################################################### script
+
+homebrew_prefix_candidates=()
+
+usage() {
+  cat <<EOS
+Homebrew Uninstaller
+Usage: $0 [options]
+    -p, --path=PATH  Sets Homebrew prefix. Defaults to /usr/local.
+        --skip-cache-and-logs
+                     Skips removal of HOMEBREW_CACHE and HOMEBREW_LOGS.
+    -f, --force      Uninstall without prompting.
+    -q, --quiet      Suppress all output.
+    -d, --dry-run    Simulate uninstall but don't remove anything.
+    -h, --help       Display this message.
+EOS
+  exit "${1:-0}"
+}
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -p*) homebrew_prefix_candidates+=("${1#-p}");;
+    --path=*) homebrew_prefix_candidates+=("${1#--path=}");;
+    --skip-cache-and-logs) opt_skip_cache_and_logs=1;;
+    -f|--force) opt_force=1;;
+    -q|--quiet) opt_quiet=1;;
+    -d|--dry-run) opt_dry_run=1;;
+    -h|--help) usage;;
+    *) warn "Unrecognized option: '$1'"; usage 1;;
+  esac
+  shift
+done
+
+if [[ ${#homebrew_prefix_candidates[@]} -eq 0 ]]; then # Attempt to locate Homebrew unless `--path` is passed
+  prefix=$(brew --prefix)
+  [[ $prefix ]] && homebrew_prefix_candidates+=("$prefix")
```

Comment:
I think it's preferable to write `-n` explicitly.

Reply:
OK.